src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -99,22 +99,22 @@ private bool IsFullyBoundProject(BindingConfiguration binding, Project project,
                 return false;
             }
 
-            var isRuleSetBound = ruleSetReferenceChecker.IsReferenced(project, solutionRuleSet);
-
+            var isRuleSetBound = ruleSetReferenceChecker.IsReferenced(project, solutionRuleSetFilePath);
             return isRuleSetBound;
         }
 
-        private bool IsSolutionBound(BindingConfiguration binding, Language language, out RuleSet solutionRuleSet, out string additionalFilePath)
+        private bool IsSolutionBound(BindingConfiguration binding, Language language, out string solutionRuleSetFilePath, out string additionalFilePath)
         {
-            solutionRuleSet = null;
+            solutionRuleSetFilePath = null;
             additionalFilePath = CSharpVBBindingConfigProvider.GetSolutionAdditionalFilePath(language, binding);
 
             if (!fileSystem.File.Exists(additionalFilePath))
             {
                 return false;
             }
 
-            solutionRuleSet = GetSolutionRuleSet(binding, language);
+            var solutionRuleSet = GetSolutionRuleSet(binding, language);
```

Comment:
Hmm, in `GetSolutionRuleSet` we actually try to load the ruleset -- do you think that's really needed? the code can be further simplified and we can change it to `GetSolutionRuleSetFilePath`

Reply:
I did try removing it, but it's still required. `GetSolutionRuleSet` calls the ruleset serializer, which checks the file exists, tries to load it, and returns null if the file can't be loaded i.e. it's checking the presence and validity of the ruleset. If the ruleset is missing or invalid we want `IsSolutionBound` to return `false`.

I thought I'd put in a comment about this but I can't find it...

Reply:
I was thinking that maybe we can remove the validity check, since we can check that the file exists using the file system, so we only serialize to check that it's valid. But on second thought, it's useful to check, so ignore my comment :)